<html>

<head>

    <meta http-equiv="Content-Type" content="text/html;"/>

    <style>

        .command-line {
            font-size: large;
            color: #990000
        }

    </style>
</head>
<body>


<h1>Computing Coverage with igvtools</h1>


<p>In this exercise we will compute total and strand-specific coverage from RNA-Seq alignments. The coverage files
    will be examined visually in IGV, and strand information compared with known
    annotations.</p>


<h2>Data</h2>


<p>Data for this tutorial was provided by Brian Haas and can be found in the &#34;data/rna&#34; directory. This dataset
    was derived from S.Pombesamples and includes RNA-Seq strand-specific paired-end reads, a genome fasta reference
    file,
    annotations, and genome annotations</p>


<h2>Prerequisites</h2>


<p>igvtools requires Java version 1.6 or later, so before proceeding verify that Java is installed
    by executing the following from a command terminal.

<p class="command-line">java -version</p>


<h2>Install igvtools</h2>


<p>Download the latest version of igvtools
    from <a href="http://www.broadinstitute.org/igv/download">www.broadinstitute.org/igv/download</a>
    and save the zip archive to your home directory. Some browsers, notably safari, will unzip
    the archive automatically. In this case just move the IGVToolsfolder to your home
    directory and skip the unzip step below.</p>


<p>Open a command terminal in your home directory and unzip the archive (excute "cd" to insure you are in the home
    directory)</p>

<p class="command-line">unzip igvtools_2.3.37.zip </p>

<p>Verify the installation by executing the following</p>

<p class="command-line">~/IGVTools/igvtools</p>

<p>You should see a version and usage string followed by a summary of commands.</p>


<h2>Set the path variable</h2>

<p>Now add the igvtools command to your path environment variable.This is optional but
    makes typing the command convenient. If you skip this step you will need to type the full path
    (or ~/IGVTools/igvtool) to execute the program. The syntax depends on the shell you are running. </p>

<p>For csh and tcsh</p>

<p class="command-line">set path = ( $path ~/IGVTools)</p>

<p>For bash:</p>

<p class="command-line">PATH=$PATH:=IGVTools</p>

<p>If you aren&#39;t sure which shell you are running try both commands above (one should work and the other is
    harmless).</p>

<p>Verify the path setting by executing</p>

<p class="command-line">igvtools</p>

<h2>Compute total and strand-specific coverage</h2>

<p>To compute a total coverage file in &#34;tdf&#34;format first open a terminal in the data/rna directory then execute
    the following:</p>

<p class="command-line">igvtools count Sp_ds.bam Sp_ds.bam.tdf genome.fa</p>

<p>This should produce a file called &#34;Sp_ds.bam.tdf&#34; containing coverage data for the entire bam.<span

<p>Now compute a strand-specific coverage file. This is possible because the RNA library used for the sample
    dataset was strand preserving. Specifically, for this library, the strand of the first read of each pair corresponds
    to the strand of
    the fragment sequenced. Use the optional &#34;--strands&#34; arugment to separate the coverage count by strand</p>

<p class="command-line">count &nbsp; --strands first Sp_ds.bam Sp_ds.bam.strands.tdf genome.fa</p>

<h2>View the coverage files in IGV</h2>

<p>Launch igv and load the custom genome, annotations, and bam file.</p>
<ul>
    <li>
        From the menu &#34;Genomes &gt; Load Genome from File...&#34;select &#34;genome.fa&#34;in the data/rna folder.
    </li>
    <li>
        From the menu &#34;File &gt; Load from File...&#34; select &#34;genes.bed&#34; in the data/rna folder
    </li>
    <li>
        From the menu &#34;File &gt; Load from File...&#34; select &#34;Sp_ds.bam&#34; in the data/rna folder
    </li>
</ul>


<p><i>Note: the total coverage file is loaded automatically.</i></p>


<p>Load the strand specific coverage file &#34;Sp_ds.bam.strands.tdf&#34;</p>

<p>Zoom in on a peak until alignments are shown</p>


<p>Color alignments by strand by right-clicking in the alignment track, then selecting &#34;Color alignments by &gt;
    first-of-pair
    strand&#34;.</p>

<p>Verify that coverage tracks and alignments correspond with the known strands of the gene annotations. </p>


</body>

</html>
